### PR TITLE
[new feature]add operation max

### PR DESCRIPTION
**Pull request by qiuxin2012**
## What changes were proposed in this pull request?
add operation max

## How was this patch tested?

unit tests

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Max.scala [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ */
+package com.intel.analytics.bigdl.nn.ops
+
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.Table
+
+import scala.reflect.ClassTag
+
+class Max[T: ClassTag, D: ClassTag](
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Max.scala]: java doc. we should explain what's it for and what's the input like, what's the output like

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Max.scala [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      ) extends Operation[Table, Tensor[D], T] {
+
+  output = Tensor[D]()
+  protected val indices: Tensor[D] = Tensor[D]()
+
+  override def updateOutput(input: Table): Tensor[D] = {
+    val x = input[Tensor[D]](1)
+    val y = input[Tensor[Int]](2)
+
+    require(y.dim() == 1 && y.nElement() == 1, s"reduction indices should be a scalar")
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Max.scala]: use isScalar method?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Max.scala [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    val reductionIndices = y.valueAt(1) + 1
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Max.scala]: If it's scalar, we may use value

Comment (yiheng) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Max.scala]: should we make the +1 configurable? What if user pass in a 1 based index

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Max.scala [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+package com.intel.analytics.bigdl.nn.ops
+
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.Table
+
+import scala.reflect.ClassTag
+
+/**
+ * Computes the maximum of elements across dimensions of a tensor.
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Max.scala]: we need to tell user what kind of input the operation expect, e.g. two tensor, the first one is data, the second one is index

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Max.scala [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+ * @param keepDims if keepDims is false, will delete the singleton dimension
+ *                 in output.
+ */
+class Max[T: ClassTag, D: ClassTag](
+        keepDims: Boolean = false,
+        startFromZero: Boolean = false
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Max.scala]: add startFromZero to the java doc

**Comment by yiheng** [comment]
LGTM. Please run jenkins

**Comment by qiuxin2012** [comment]
jenkins passed http://mlt-bddl01.sh.intel.com:8080/view/PR/job/BigDL-PRVN-lite/9/

**Pull request merged**